examples/stdlib/stream/zip.effekt

```diff
@@ -0,0 +1,14 @@
+import stream
+import array
+
+def main() = {
+  def stream1() = rangeFrom(1)
+  // Ideally this would be an array to demonstrate the capabilities. 
+  def stream2() = ['H', 'e', 'l', 'l', 'o'].each
```

Comment:
I'd love this to be an array like:
```suggestion
  // Ideally this would be an array to demonstrate the capabilities. 
  def stream2() = array::fromList(['H', 'e', 'l', 'l', 'o']).each
```
but it runs into the classic Valgrind issue we discussed at a meeting.

Reply:
Or a string like:
```suggestion
  // Ideally this would be an array to demonstrate the capabilities. 
  def stream2() = "Hello".each
```
but it doesn't seem to work the way I would expect, then the zipping returns:
```
1, H
2, H
3, H
4, H
...
```
(an infinite stream of `($number, 'H')`